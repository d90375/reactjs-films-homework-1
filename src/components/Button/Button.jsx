import React from 'react';
import PropTypes from 'prop-types';

import styles from './Button.scss';

const Button = ({ children, color, onClick }) => (
  <button
    className={`${styles.button} ${styles[color]}`}
    type="button"
    onClick={onClick}
  >
    {children}
  </button>
);

Button.propTypes = {
  children: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Button;
