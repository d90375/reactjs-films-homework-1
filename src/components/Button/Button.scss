@import '../../mixins.scss';

.button {
  min-width: 100px;
  padding: 0 10px;
  line-height: 30px;
  font-size: 16px;
  color: #fff;
  border-radius: 5px;
  cursor: pointer;

  @include outline-none();
}

.primary {
  background-color: #0aaee4;
  border: none;

  &:hover {
    background-color: #0aade4be;
  }
}

.secondary {
  background-color: transparent;
  border: 2px solid #fff;

  &:hover {
    background-color: #ffffff38;
  }
}
