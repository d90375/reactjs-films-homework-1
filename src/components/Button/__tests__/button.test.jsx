import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Button from '../Button';

test('ButtonView renders correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Button color="blue" onClick={() => {}}>child</Button>);
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
