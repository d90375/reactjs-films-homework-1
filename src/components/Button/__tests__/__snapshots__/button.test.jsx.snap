// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonView renders correctly 1`] = `
<button
  className="button blue"
  onClick={[Function]}
  type="button"
>
  child
</button>
`;
